src/runtime/plugins/types.d.ts

```diff
@@ -1,30 +1,94 @@
 import type { Ref } from 'vue'
 import type { UnwrapNestedRefs } from 'vue'
 
+/**
+ * Result of the native PWA installation prompt.
+ */
 export interface UserChoice {
+  /**
+   * The user's selection on the install prompt.
+   */
   outcome: 'accepted' | 'dismissed'
+  /**
+   * Platform identifier provided by the browser for where the prompt was shown
+   * (for example, 'web' or a store-specific value). The exact values are
+   * browser-dependent and not standardized.
+   */
   platform: string
 }
 
+/**
+ * Browser event fired before showing the native PWA install prompt.
+ *
+ * This event allows delaying the native prompt and showing a custom UI first.
+ */
 export type BeforeInstallPromptEvent = Event & {
+  /**
+   * Triggers the browser's native install prompt.
+   */
   prompt: () => void
+  /**
+   * Resolves with the user's selection once they interact with the prompt.
+   */
   userChoice: Promise<UserChoice>
 }
 
+/**
+ * Reactive PWA state and helpers injected into Nuxt as `$pwa`.
+ */
 export interface PwaInjection {
   /**
-   * @deprecated use `isPWAInstalled` instead
+   * @deprecated Use `isPWAInstalled` instead. Legacy boolean computed from
+   * browser heuristics to detect if the app is installed.
    */
   isInstalled: boolean
+  /**
+   * Whether the app is currently installed as a PWA.
+   * This value updates as installation state changes (e.g. display-mode changes).
+   */
   isPWAInstalled: Ref<boolean>
+  /**
+   * When `true`, your UI should show a custom install prompt. This flag is set
+   * after the `beforeinstallprompt` event is captured and cleared when the user
+   * proceeds or cancels.
+   */
   showInstallPrompt: Ref<boolean>
+  /**
+   * Cancels the custom install flow and hides future prompts by setting an
+   * opt-out flag (stored in `localStorage` when configured via `installPrompt`).
+   */
   cancelInstall: () => void
+  /**
+   * Shows the native install prompt if available and returns the user's choice.
+   * Returns `undefined` when the prompt is not available or not currently shown.
+   */
   install: () => Promise<UserChoice | undefined>
+  /**
+   * Whether the service worker has reached the `activated` state.
+   */
   swActivated: Ref<boolean>
+  /**
+   * Indicates that registering the service worker failed.
+   */
   registrationError: Ref<boolean>
+  /**
+   * Becomes `true` when the app is ready to work offline (precache completed).
+   */
```

Comment:
This flag is activated only once, when the service worker is registered and activated for first time.